src/corelibs/U2Gui/src/util/CreateAnnotationWidgetController.cpp

```diff
@@ -120,14 +120,17 @@ CreateAnnotationWidgetController::CreateAnnotationWidgetController(const CreateA
 
     commonWidgetUpdate(model);
 
-    connect(w, SIGNAL(si_selectExistingTableRequest()), SLOT(sl_onLoadObjectsClicked()));
-    connect(w, SIGNAL(si_selectGroupNameMenuRequest()), SLOT(sl_groupName()));
-    connect(w, SIGNAL(si_groupNameEdited()), SLOT(sl_groupNameEdited()));
-    connect(w, SIGNAL(si_annotationNameEdited()), SLOT(sl_annotationNameEdited()));
-    connect(w, SIGNAL(si_usePatternNamesStateChanged()), SLOT(sl_usePatternNamesStateChanged()));
+    connect(w.data(), SIGNAL(si_selectExistingTableRequest()), SLOT(sl_onLoadObjectsClicked()));
+    connect(w.data(), SIGNAL(si_selectGroupNameMenuRequest()), SLOT(sl_groupName()));
+    connect(w.data(), SIGNAL(si_groupNameEdited()), SLOT(sl_groupNameEdited()));
+    connect(w.data(), SIGNAL(si_annotationNameEdited()), SLOT(sl_annotationNameEdited()));
+    connect(w.data(), SIGNAL(si_usePatternNamesStateChanged()), SLOT(sl_usePatternNamesStateChanged()));
     connect(occ, SIGNAL(si_comboBoxChanged()), SLOT(sl_documentsComboUpdated()));
 }
 
+CreateAnnotationWidgetController::~CreateAnnotationWidgetController() {
```

Comment:
Remove the empty destructor

Reply:
If I remove the empty destructor, the build fails because the default destructor is inline, but to declare a QScopedPointer to a forward declared class, it must be non-inline https://doc.qt.io/qt-5/qscopedpointer.html#forward-declared-pointers

I can leave a comment with a link that this empty destructor cannot be deleted